src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1340,12 +1335,74 @@ public static bool HasIdentityConversionToAny<T>(T type, ArrayBuilder<T> targetT
         public Conversion ConvertExtensionMethodThisArg(TypeSymbol parameterType, TypeSymbol thisType, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
             Debug.Assert((object)thisType != null);
-            var conversion = this.ClassifyImplicitConversionFromType(thisType, parameterType, ref useSiteDiagnostics);
+            var conversion = this.ClassifyImplicitExtensionMethodThisArgConversion(null, thisType, parameterType, ref useSiteDiagnostics);
             return IsValidExtensionMethodThisArgConversion(conversion) ? conversion : Conversion.NoConversion;
         }
 
         // Spec 7.6.5.2: "An extension method ... is eligible if ... [an] implicit identity, reference,
         // or boxing conversion exists from expr to the type of the first parameter"
+        public Conversion ClassifyImplicitExtensionMethodThisArgConversion(BoundExpression sourceExpressionOpt, TypeSymbol sourceType, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+        {
+            Debug.Assert(sourceExpressionOpt == null || (object)sourceExpressionOpt.Type == sourceType);
+            Debug.Assert((object)destination != null);
+
+            if ((object)sourceType != null)
+            {
+                if (HasIdentityConversion(sourceType, destination))
+                {
+                    return Conversion.Identity;
+                }
+
+                if (HasBoxingConversion(sourceType, destination, ref useSiteDiagnostics))
+                {
+                    return Conversion.Boxing;
+                }
+
+                if (HasImplicitReferenceConversion(sourceType, destination, ref useSiteDiagnostics))
+                {
+                    return Conversion.ImplicitReference;
+                }
+            }
+
+            if (sourceExpressionOpt?.Kind == BoundKind.TupleLiteral)
+            {
+                var tupleConversion = GetTupleLiteralConversion(
+                    (BoundTupleLiteral)sourceExpressionOpt,
+                    destination,
+                    ref useSiteDiagnostics,
+                    ConversionKind.ImplicitTupleLiteral,
+                    (BoundExpression s, TypeSymbol d, ref HashSet<DiagnosticInfo> u, bool a) => ClassifyImplicitExtensionMethodThisArgConversion(s, s.Type, d, ref u),
+                    arg: false);
+                if (tupleConversion.Exists)
+                {
+                    return tupleConversion;
+                }
+            }
+
+            if ((object)sourceType != null)
+            {
+                var tupleConversion = ClassifyTupleConversion(
```

Comment:
We don't know we have a tuple at this point but `ClassifyTupleConversion` returns `NoConversion` if either the source or destination is not a tuple. This follows the pattern used elsewhere in `ConversionsBase`. See the call to `ClassifyImplicitTupleConversion` in `ClassifyStandardImplicitConversion` for instance.